core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerService.scala

```diff
@@ -324,7 +324,9 @@ class LoadBalancerService(config: WhiskConfig, instance: InstanceId, entityStore
 
     loadBalancerData.activationCountPerInvoker.flatMap { currentActivations =>
       allInvokers.flatMap { invokers =>
-        val invokersToUse = if (action.exec.pull) blackboxInvokers(invokers) else managedInvokers(invokers)
+        val healthyInvokers = invokers.filter(_._2 == Healthy)
+        val invokersToUse =
+          if (action.exec.pull) blackboxInvokers(healthyInvokers) else managedInvokers(healthyInvokers)
```

Comment:
Keeping Unhealthy/Down invokers in that list is deliberate:

The loadbalancer heavily relies on the ordering of invokers. If You shrink/grow this list continually (for example by an invoker going online/offline all the time) you completely destroy its heuristics, resulting in super bad scheduling behavior if you have a system with some invokers in a flipping state.

What's the motivation to do this change?

Reply:
currently if you deploy a single invoker, and it is not id 0, it is ignored; so while "ordering of invokers" may be important, it will require some way to make sure there is also a way to ignore invokers that should not be referenced in the scheduling. How should that be done, if not by referencing the healthy state?